app/src/features/Batches/BatchResourceDialog.tsx

```diff
@@ -0,0 +1,238 @@
+      const resourcesToKeep = selectedResourceIds.filter(
+        (id) => !displayedResourcesIds?.includes(id)
+      );
+      setSelectedResourceIds(resourcesToKeep);
+    } else if (displayedResourcesIds) {
+      setSelectedResourceIds(selectedResourceIds.concat(displayedResourcesIds));
+    }
+  };
+
+  const handleBatchRun = () => {
+    const batchCreate = async () => {
+      try {
+        await apiBatchCreate({
+          batchRequest: {
+            resources: selectedResourceIds,
+          },
+        }).unwrap();
+      } catch (e) {
+        setAlert(e.message as string);
+      }
+    };
+
+    if (selectedResourceIds.length > 0) {
+      batchCreate();
+      handleClose(false);
+      setSelectedResourceIds([]);
+    }
+  };
+  useEffect(() => {
+    setDisplayedResources(resources);
+  }, [resources]);
+
+  return (
+    <Dialog
+      open={open}
+      onClose={handleCloseModal}
+      classes={{ paper: classes.dialog }}
+      fullWidth
+    >
+      <DialogTitle className={classes.title} disableTypography>
+        <Typography variant="h6">{t("selectResources")}</Typography>
+        <div
+          onClick={handleSelectAllClick}
+          className={clsx(classes.title, classes.checkboxForm)}
+        >
+          <Typography>
+            {areAllDisplayedResourcesSelected || isCheckboxIndeterminate
+              ? t("unselectResources")
+              : t("selectAllResources")}
+          </Typography>
+          <Checkbox
+            disableFocusRipple
+            edge="end"
+            color="primary"
+            checked={areAllDisplayedResourcesSelected}
+            indeterminate={isCheckboxIndeterminate}
+          />
+        </div>
+      </DialogTitle>
+      <TextField
+        variant="outlined"
+        size="small"
+        classes={{ root: classes.textField }}
+        fullWidth
+        placeholder={t<string>("search")}
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <Search />
+            </InputAdornment>
+          ),
+        }}
+        onChange={handleSearchResource}
+      />
+      <DialogContent>
+        <List>
+          {displayedResources &&
+            displayedResources.map((resource) => (
+              <BatchResourceListItem
+                resource={resource}
+                handleClick={handleSelectResources}
+                key={resource.id}
+                checked={selectedResourceIds.includes(resource.id)}
+              />
+            ))}
+        </List>
+      </DialogContent>
+      <Divider />
+      <DialogActions>
+        <Button
+          color="primary"
+          variant="contained"
+          size="large"
+          onClick={handleBatchRun}
+          disabled={selectedResourceIds.length === 0}
+        >
+          {`${t("runOn")} ${selectedResourceIds.length.toString()} ${t(
+            "resources"
+          ).toLowerCase()}`}
```

Comment:
You can confiture `t` function to take parameters like this : 

```suggestion
          {t("runOnResources", {resourceCount: selectedResourceIds.length})}
```

Check this link to know more about interpolation : https://www.i18next.com/translation-function/interpolation

Reply:
I don't know why I thought it wasn't updating when changing, but it did, my mistake